src/commands/auth/logout.ts

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { flags, FlagsConfig, SfdxCommand } from '@salesforce/command';
+import {
+  Aliases,
+  AuthInfo,
+  AuthInfoConfig,
+  Config,
+  ConfigAggregator,
+  Global,
+  Logger,
+  Messages,
+  Mode,
+  SfdxError
+} from '@salesforce/core';
+import { AsyncCreatable } from '@salesforce/kit';
+import { Nullable } from '@salesforce/ts-types';
+import * as os from 'os';
+import * as path from 'path';
+import { AuthLib } from '../../authLib';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-auth', 'logout');
+
+export default class Logout extends SfdxCommand {
+  public static readonly description = messages.getMessage('description');
+  public static readonly examples = messages
+    .getMessage('examples')
+    .split(os.EOL);
+  public static readonly supportsUsername = true;
+  public static readonly flagsConfig: FlagsConfig = {
+    all: flags.boolean({
+      char: 'a',
+      description: messages.getMessage('all'),
+      longDescription: messages.getMessage('allLong'),
+      required: false
+    }),
+    noprompt: flags.boolean({
+      char: 'p',
+      description: messages.getMessage('noPrompt'),
+      required: false
+    })
+  };
+
+  async run(): Promise<string[]> {
+    if (this.flags.targetusername && this.flags.all) {
+      const err = new SfdxError(
+        messages.getMessage('specifiedBothUserAndAllError'),
+        'SpecifiedBothUserAndAllError'
+      );
+      return Promise.reject(err);
+    }
+    const helper = await LogoutHelper.create({
+      targetUsername: this.flags.targetusername,
+      all: this.flags.all
+    });
+
+    if (await this.shouldRunCommand(helper.authConfigs)) {
+      const usernames = [...helper.authConfigs.keys()];
+      for (const username of usernames) {
+        await helper.removeAuth(username);
+      }
+      this.ux.log(
+        messages.getMessage('logoutOrgCommandSuccess', [usernames.join(os.EOL)])
+      );
+      return usernames;
+    }
+    return [];
+  }
+
+  private async shouldRunCommand(
+    authConfigs: LogoutHelper.AuthConfigs
+  ): Promise<boolean> {
+    if (this.flags.noprompt || Global.getEnvironmentMode() === Mode.DEMO) {
+      return true;
+    } else {
+      const orgsToDelete = [[...authConfigs.keys()].join(os.EOL)];
+      const answer = await this.ux.prompt(
+        messages.getMessage('logoutCommandYesNo', orgsToDelete)
+      );
+      return answer.toUpperCase() === 'YES' || answer.toUpperCase() === 'Y';
+    }
+  }
+}
+
+export class LogoutHelper extends AsyncCreatable<LogoutHelper.Options> {
```

Comment:
Doesn't feel right to have this in a command.  Seems like this (or some form of this) can be moved to the core lib.